internal/services/gcp_reservation_service.go

```diff
@@ -103,15 +104,22 @@ func CreateGCPReservation(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Get Image Name
-	name, ibErr := IBClient.GetGCPImageName(r.Context(), reservation.ImageID)
-	if ibErr != nil {
-		renderError(w, r, payloads.NewClientError(r.Context(), ibErr))
-		return
+	// Validate image
+	var name string
+	if strings.HasPrefix(payload.ImageID, "composer-api") {
+		// Composer-built image
+		name, ibErr = IBClient.GetGCPImageName(r.Context(), reservation.ImageID)
+		if ibErr != nil {
+			renderError(w, r, payloads.NewClientError(r.Context(), ibErr))
+			return
+		}
+
+		logger.Trace().Msgf("Image Name is %s", name)
+	} else {
+		// Direct image ID (e.g. from https://imagedirectory.cloud)
+		name = payload.ImageID
```

Comment:
Why? I want the name, not the id. 
for example, the image id I know is e85c829f-bb14-4659-9571-25df69c0365a and the image name will not be found.

Reply:
if this would be UUID, it would fall in the other branch and got resolved by IB client